core/injector/application_injector.cpp

```diff
@@ -1147,7 +1154,16 @@ namespace {
           return get_extrinsic_observer_impl(injector);
         }),
         di::bind<authority::AuthorityUpdateObserver>.template to<authority::AuthorityManagerImpl>(),
-        di::bind<authority::AuthorityManager>.template to<authority::AuthorityManagerImpl>(),
+        bind_by_lambda<authority::AuthorityManager>([](auto const &injector) {
+          auto auth_manager_impl =
+              injector.template create<sptr<authority::AuthorityManagerImpl>>();
+          auto block_tree_impl =
+              injector.template create<sptr<blockchain::BlockTree>>();
+          auto justification_storage_policy = injector.template create<
+              sptr<blockchain::JustificationStoragePolicyImpl>>();
+          justification_storage_policy->initBlockchainInfo(block_tree_impl);
```

Comment:
I think we can do it in prepare phase of which called by AppStateManager

Reply:
Hm, but we'd have to inject it initially via DI to do it? I'm afraid this will cause circular dependency problem this method is meant to resolve.